web/modules/custom/solution/solution.module

```diff
@@ -177,6 +226,34 @@ function solution_entity_extra_field_info() {
 }
 
 /**
+ * Implements hook_ENTITY_TYPE_load().
+ */
+function solution_rdf_entity_load($entities) {
+  /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+  foreach ($entities as $entity) {
+    if ($entity->bundle() != 'solution') {
+      return;
+    }
+    // In ADMS-AP collections point to solutions.
+    // The reverse relation would have been more logical,
+    // and this is quite inconvenient, especially for the search index.
+    // To work around this, we add a computed field here.
+    $ids = \Drupal::entityQuery('rdf_entity')
+      ->condition('rid', 'collection')
+      ->condition('field_ar_affiliates', $entity->id())
+      ->execute();
```

Comment:
I'm a bit concerned about the performance impact here. This does an additional query every time a solution is loaded, regardless of whether this data is even needed.

Maybe make a follow up issue to turn this into an entity reference field + a `hook_entity_update()` implementation that keeps the field updates whenever a reverse reference changes?